restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -803,28 +798,70 @@ void addRowsToMap(Map<String, List<Row>> rowsByDoc, List<Row> rows) {
     return Collections.emptyList();
   }
 
+  /**
+   * If the DocumentSearchPageState has a defined last seen `documentId`, advance the result set to
+   * be just after that ID.
+   */
+  private List<Row> skipSeenRows(DocumentSearchPageState initialPagingState, List<Row> rows) {
+    if (initialPagingState == null) {
+      return rows;
+    }
+
+    String lastSeenId = initialPagingState.getLastSeenDocId();
+    if (lastSeenId == null) {
+      return rows;
+    }
+
+    boolean idFound = false;
+    boolean doneSkipping = false;
+    while (!doneSkipping && rows.size() > 0) {
+      String docId = rows.get(0).getString("key");
+      if (docId.equals(lastSeenId)) {
+        idFound = true;
+        rows.remove(0);
+      } else if (idFound) {
+        doneSkipping = true;
+      } else {
+        rows.remove(0);
+      }
+    }
+
+    return rows;
```

Comment:
oops, looks like `dropWhile` is not available java8. :(

Reply:
Other suggestion could be to create a new List and pass it:
```
boolean idFound = false;
List<Row> newRows = new ArrayList<>();
for (Row row : rows) {
     if (docId.equals(lastSeenId)) {
         found = true;
         continue;
     }
     if (found) {
         newRows.add(row);
     }
}
return newRows;
```

Reply:
@EricBorczuk Instead of skipping by changing the list in place wouldn't be simpler to create a copy? Just in case you didn't see this snippet above. :)